udata/harvest/backends/dcat.py

```diff
@@ -146,25 +156,51 @@ def walk_graph(self, url: str, fmt: str) -> Generator[tuple[int, Graph], None, N
                     break
 
             yield page_number, subgraph
-            if self.is_done():
+            if self.has_reached_max_items():
                 return
 
             page_number += 1
 
     def process_one_datasets_page(self, page_number: int, page: Graph):
         for node in page.subjects(RDF.type, DCAT.Dataset):
             remote_id = page.value(node, DCT.identifier)
+            if self.is_dataset_external_to_this_page(page, node):
+                continue
+
             self.process_dataset(remote_id, page_number=page_number, page=page, node=node)
 
-            if self.is_done():
+            if self.has_reached_max_items():
                 return
 
+    def is_dataset_external_to_this_page(self, page: Graph, node) -> bool:
```

Comment:
Let's check if the title is missing as well to make sure that we don't lose datasets that were currently harvested

Reply:
Done in 604e3c2f0